pgtype/bool.go

```diff
@@ -309,9 +316,28 @@ func (scanPlanTextAnyToBoolScanner) Scan(src []byte, dst any) error {
 		return s.ScanBool(Bool{})
 	}
 
-	if len(src) != 1 {
-		return fmt.Errorf("invalid length for bool: %v", len(src))
+	if len(src) == 0 {
+		return fmt.Errorf("cannot scan empty string into %T", dst)
 	}
 
-	return s.ScanBool(Bool{Bool: src[0] == 't', Valid: true})
+	v, err := planTextToBool(src)
+	if err != nil {
+		return err
+	}
+
+	return s.ScanBool(Bool{Bool: v, Valid: true})
+}
+
+// https://www.postgresql.org/docs/11/datatype-boolean.html
+func planTextToBool(src []byte) (bool, error) {
+	s := string(bytes.ToLower(bytes.TrimSpace(src)))
+
+	switch {
+	case strings.HasPrefix("true", s), strings.HasPrefix("yes", s), s == "on", s == "1":
```

Comment:
This will match `t`, `tr`, `tru`, `true`, is this by purpose?

Reply:
From docs:
> Unique prefixes of these strings are also accepted